packages/compiler/core/types.ts

```diff
@@ -151,7 +151,16 @@ export interface ModelType extends BaseType, DecoratedType, TemplatedType {
     | ProjectionModelExpressionNode;
   namespace?: NamespaceType;
   properties: Map<string, ModelTypeProperty>;
+
+  /**
+   * Model this model extends. This represent inheritance.
+   */
   baseModel?: ModelType;
+
+  /**
+   * Direct children. This is the reverse relation of @see baseModel
+   */
+  children?: ModelType[];
```

Comment:
I would name this `childModels` to better align with the `baseModel` naming convention.

Can we also make it required? Empty vs non-empty array is easier to deal with generally.

Reply:
I don't feel strongly about this, but I've been wondering if we should avoid "child" to refer to inheritance. I think I would prefer "derived" or "sub". The problem with "child", to me, is that it can be ambiguous. And syntactically at least, we have a "parent" node that is not related to the base type.

Reply:
I don't feel strongly about this, but I've been wondering if we should avoid "child" to refer to inheritance. I think I would prefer "derived" or "sub". The problem with "child", to me, is that it can be ambiguous. And syntactically at least, we have a "parent" node that is not related to the base type.

Reply:
(Unresolving only to show this non-blocking observation)

Reply:
I can see this being confusing, should we add that to the syntax review?

Reply:
Yeah that is true, let me check if I can find other naming in other compilers

Reply:
Typescript compiler does seem to use the term `derived` for this. Does also looks like that in csharp compiler. DOes `derivedModels` sounds like a good option?

Reply:
Fine with derivedModels